File: easy/go/0013.go
func romanToInt(s string) int {
    var table = map[byte]int{'I':1, 'V':5, 'X':10, 'L':50, 'C':100, 'D':500, 'M':1000}
 	var number int = 0
 	for i := 0 ; i < len(s) - 1;i++ {
        if table[s[i]] >= table[s[i+1]]{
            number += table[s[i]]
        }else{
            number -= table[s[i]]
        }
 	}
    number += table[s[len(s)-1]]
	return number
}

File: easy/python/0013.py
class Solution:
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        table = {
            'I':1,
            'IV':4,
            'V':5,
            'IX':9,
            'X':10,
            'XL':40,
            'L':50,
            'XC':90,
            'C':100,
            'CD':400,
            'D':500,
            'CM':900,
            'M':1000
        }
        roman_list = ['M','CM','D','CD','C','XC','L','XL','X','IX','V','IV','I']
        number = 0
        while len(s)!=0:
            candidate = s[:2]
            if candidate in table:
                number += table[candidate]
                s = s[2:]
            elif candidate[0] in table:
                number += table[candidate[0]]
                s = s[1:]
        return number
                